### ISSUE TITLE
Appservices are not queried for users during profile requests

### ISSUE THREAD

**Issue by turt2live** [matrix-org/synapse#4310]
Related: #4143 

When Synapse receives an event that is directed at/sent by an appservice user, it ensures the appservice has registered the user by hitting the `/users` appservice API. This check is not done during profile requests, however. This means that when clients try and do profile lookups before invites (such as in the case of riot: https://github.com/vector-im/riot-web/issues/7922), they might not bother with the invite because the user doesn't exist.

TLDR: Synapse should be hitting an appservice's `/users` endpoint during profile lookup requests too.
